src/api/cli/integration/integration_test.go

```diff
@@ -1523,8 +1519,186 @@ func TestLiveSend(t *testing.T) {
+			},
+		},
+		{
+			// Send 1 coin from second to the the third address, this will spend three outputs(0.2, 0.3. 0.5 coin),
+			// and burn out the remaining 1 coin hour.
+			name: "send to burn all coin hour",
+			args: func() []string {
+				return []string{"createRawTransaction", "-a", w.Entries[1].Address.String(),
+					w.Entries[2].Address.String(), "1"}
+			},
+			checkTx: func(t *testing.T, txid string) {
+				// Confirms that the third address has 1 coin and 0 coin hour
+				coins, hours := getAddressBalance(t, w.Entries[2].Address.String())
+				require.Equal(t, uint64(1e6), coins)
+				require.Equal(t, uint64(0), hours)
+			},
+		},
+		{
+			// Send with 0 coin hour, this test should fail.
+			name: "send 0 coin hour",
+			args: func() []string {
+				return []string{"createRawTransaction", "-a", w.Entries[2].Address.String(),
+					w.Entries[1].Address.String(), "1"}
+			},
+			errMsg:  []byte("ERROR: Transaction has zero coinhour fee. See 'skycoin-cli createRawTransaction --help'"),
+			checkTx: func(t *testing.T, txid string) {},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create raw transaction first
+			output, err := exec.Command(binaryPath, tc.args()...).CombinedOutput()
+			if err != nil {
+				t.Fatalf("err: %v, output: %v", err, string(output))
+				return
+			}
+			require.NoError(t, err)
+			output = bytes.TrimRight(output, "\n")
+			if bytes.Contains(output, []byte("ERROR:")) {
+				require.Equal(t, tc.errMsg, output)
+				return
+			}
+
+			// Broadcast transaction
+			output, err = exec.Command(binaryPath, "broadcastTransaction", string(output)).CombinedOutput()
+			require.NoError(t, err)
+
+			txid := string(bytes.TrimRight(output, "\n"))
+			fmt.Println("txid:", txid)
+			_, err = cipher.SHA256FromHex(txid)
+			require.NoError(t, err)
+
+			// Wait untill transaction is confirmed.
+			tk := time.NewTicker(time.Second)
+		loop:
+			for {
+				select {
+				case <-time.After(30 * time.Second):
+					t.Fatal("Wait tx confirmation timeout")
+				case <-tk.C:
+					if isTxConfirmed(t, txid) {
+						break loop
+					}
+				}
+			}
+
+			tc.checkTx(t, txid)
+		})
+	}
+
+	// Send with too small decimal value
+	errMsg := []byte("ERROR: invalid amount, too many decimal places. See 'skycoin-cli createRawTransaction --help'")
+	for i := 0.0001; i <= 0.0009; i++ {
+		v, err := droplet.ToString(uint64(i * droplet.Multiplier))
+		require.NoError(t, err)
+		name := fmt.Sprintf("send %v", v)
+		t.Run(name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, "createRawTransaction", w.Entries[0].Address.String(), v).CombinedOutput()
+			require.NoError(t, err)
+			output = bytes.Trim(output, "\n")
+			require.Equal(t, errMsg, output)
+		})
+	}
+}
+
 func getTransaction(t *testing.T, txid string) *webrpc.TxnResult {
 	output, err := exec.Command(binaryPath, "transaction", txid).CombinedOutput()
+	if err != nil {
+		fmt.Println(string(output))
+		return &webrpc.TxnResult{}
+	}
```

Comment:
fail the test only?

Reply:
No, we don't fail here, It's the error that the transaction doesn't exist, we only need to check again next time.

Reply:
Oh okay 👌